components/gfx/paint_context.rs

```diff
@@ -909,12 +909,37 @@ impl<'a> PaintContext<'a> {
         }
 
         // FIXME(pcwalton): This surface might be bigger than necessary and waste memory.
-        let size = self.draw_target.get_size();
-        let size = Size2D(size.width, size.height);
+        let size = self.draw_target.get_size(); //Az size.
+        let mut size = Size2D(size.width, size.height); //Geom::Size.
+
+        // Pre-calculate if there is a blur expansion need.
+        let accum_blur = filters::calculate_accumulated_blur(filters);
+        let mut matrix = Matrix2D::identity();
+        if accum_blur > Au::new(0) {
+            // Set the correct size.
+            let rect = Rect(Point2D(0, 0), size);
+            let side_inflation = accum_blur * BLUR_INFLATION_FACTOR;
+            let inflated_size = rect.inflate(side_inflation.to_nearest_px() as i32, side_inflation.to_nearest_px() as i32);
+            size = inflated_size.size;
```

Comment:
If you're just using the size part of the rect, why use a rect and `inflate()` at all? Isn't `size = Size2D((side_inflation.to_nearest_px() * 2) as i32, (side_inflation.to_nearest_px() * 2) as i32` enough?